viz_scripts/scaffolding.py

```diff
@@ -161,6 +174,31 @@ def load_viz_notebook_data(year, month, program, study_type, dic_re, dic_pur=Non
 
     return expanded_ct, file_suffix, quality_text, debug_df
 
+# Function to map the "MODE", "REPLACED_MODE", "PURPOSE" to respective en-translations
+# Input: dynamic_labels, label_type: MODE, REPLACED_MODE, PURPOSE
+# Return: Dictionary mapping between the label type and its english translation.
+def mapping_labels(dynamic_labels, label_type):
+    if "translations" in dynamic_labels and "en" in dynamic_labels["translations"]:
+        translations = dynamic_labels["translations"]["en"]
+        dic_mapping = dict()
+
+        def translate_labels(labels):
+            translation_mapping = {}
+            for label in labels:
+                value = label["value"]
+                translation = translations.get(value)
+                translation_mapping[value] = translation
+            return defaultdict(lambda: 'Other', translation_mapping)
+
+        if (label_type == "MODE"):
+            dic_mapping = translate_labels(dynamic_labels[label_type])
+        elif (label_type == "REPLACED_MODE"):
+            dic_mapping = translate_labels(dynamic_labels[label_type])
+        elif (label_type == "PURPOSE"):
+            dic_mapping = translate_labels(dynamic_labels[label_type])
```

Comment:
why do we need this if/else block when all the cases are the same?

Reply:
If/else block are redundant in this case. These have been removed.